src/main/java/uk/gov/register/configuration/ConfigManager.java

```diff
@@ -0,0 +1,48 @@
+package uk.gov.register.configuration;
+
+import uk.gov.register.RegisterConfiguration;
+
+import java.io.IOException;
+import java.net.URL;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.Optional;
+
+public class ConfigManager {
+    private final RegisterConfiguration registerConfiguration;
+    private final Optional<String> registersConfigFileUrl;
+    private final String registersConfigFilePath;
+    private final Optional<String> fieldsConfigFileUrl;
+    private final String fieldsConfigFilePath;
+    private final boolean refresh;
+    private final String externalConfigDirectory;
+
+    public ConfigManager(RegisterConfiguration registerConfiguration, Optional<String> registersConfigFileUrl, Optional<String> fieldsConfigFileUrl) {
```

Comment:
If you inject a `RegisterConfigConfiguration` rather than a whole `RegisterConfiguration`, this class will need less setup code in order to test